students/itsabhi/Lesson4/codekata.py

```diff
@@ -0,0 +1,70 @@
+#! /usr/bin/env python3
+
+
+def wordmap(fh):
+    # This function creates a dictionary of trigram sequence
+    oneline = ""
+    while True:
+        line = fh.readline()
+        oneline += line.rstrip('\r')
+        if not line:
+            break
+
+    print(oneline)
+    word_list = oneline.split()
+    n = 0
+    word_dict = {}
+    while(n+2 < len(word_list)):
+        f_word = word_list[n]
+        s_word = word_list[n+1]
+        t_word = word_list[n+2]
+
+        if (f_word + " " + s_word) not in word_dict:  # If key exists in dict
+            #Handling period, bang or question mark
+            if s_word.endswith("." or "!" or "?"):
+                punctuation = s_word[-1]
+            else:
+                punctuation = ""
```

Comment:
Check the built in `str.replace()` or `str.translate()` methods for a way to simplify the punctuation removal and apply it to the entire text so you don't end up with repeated code here